src/typing/type_inference_js.ml

```diff
@@ -1292,8 +1292,21 @@ and toplevels cx stmts =
       if uc < List.length stmts
       then (
         let msg = "unreachable code" in
-        let (loc, _) = List.nth stmts uc in
-        Flow_js.add_warning cx [mk_reason "" loc, msg]
+        let rec drop n lst =
+          match (n, lst) with
+          | (_, []) -> []
+          | (0, l) -> l
+          | (x, h :: t) -> drop (x-1) t
+        in
+        let trailing = drop uc stmts in
+        trailing |> List.iter Ast.Statement.(fun stmt ->
```

Comment:
> (could do something similar with drop by passing (n, lst) instead of n lst)

It seems nicer to keep it in curried form, but if there are curry/uncurry functions I could have the best of both worlds. I'll have a look into that now.